lib/src/session.rs

```diff
@@ -67,6 +67,14 @@ pub struct Session {
     ///
     /// Populated prior to guest execution, and never modified.
     pub(crate) backends: Arc<Backends>,
+    /// The dictionaries configured for this execution.
+    ///
+    /// Populated prior to guest execution, and never modified.
+    pub(crate) dictionaries: Arc<Dictionaries>,
+    /// The dictionaries configured for this execution.
+    ///
+    /// Populated prior to guest execution, and never modified.
+    dictionaries_by_name: PrimaryMap<DictionaryHandle, DictionaryName>,
```

Comment:
Suggestion: I think we can just turn this into:

```rust
dictionaries: PrimaryMap<DictionaryHandle, Dictionary>
```

and load it up as we construct the session below. That would eliminate the need for nested lookup in `dictionary`, and since `Dictionary` contains the name we should still be able to locate it just fine. It would also help clarify the naming here, since `dictionaries_by_name` sounds like a map that keys by name.

wdyt?

Reply:
https://github.com/fastly/Viceroy/pull/61#discussion_r686123479

(_whispering_) this was me!

Reply:
Ahha, apologies for the inconsistent direction here!

I'll note that right now the code is preloading the `dictionaries_by_name` map when creating the session, rather than lazily allocating as you open them, which is why it seemed like we could just nix the other map. Either way is fine by me!

Reply:
Ah! I would tilt towards keeping the map, because it would lead to inconsistent behavior between the production platform and Viceroy. Today, if you call `Dictionary::open("some_name")` twice, you'll be given back two handles with distinct inner `0`, `1` u32 values.

That relates to some platform limits concerning how many dictionaries can be "opened" in a session, so it feels worth keeping the lazy handle-to-name mapping.

Reply:
Excellent point @cratelyn. So for clarity, @JakeChampion that means we should move away from the preloading currently being done during `Session` construction, and instead blindly allocate a new handle every time `open` is called (no need for searching by name).

(I could see this behavior changing in C@E eventually, but best to match for now.)

Reply:
I've changed this to now allocate handles on demand -- [`edfe5a8` (#61)](https://github.com/fastly/Viceroy/pull/61/commits/edfe5a810dbb1c042fc12b297f6ebb93409105cf)